pkg/config/defaults.go

```diff
@@ -0,0 +1,257 @@
+	// So we should disable the feature with an option "--fail-swap-on=false".
+	if !cfg.DevCluster && utils.CheckVersionConstraint(cfg.KubernetesVersion, "<1.8.0") {
+		cfg.RuntimeConfiguration.FailSwapOn = true
+	}
+
+	var err error
+	switch cfg.ContainerRuntime {
+	case RuntimeDocker:
+		err = SetDefaults_DockerRuntime(cfg)
+	case RuntimeRkt:
+		err = SetDefaults_RktRuntime(cfg)
+	case RuntimeCrio:
+		err = SetDefaults_CrioRuntime(cfg)
+	}
+	if err != nil {
+		return err
+	}
+
+	// NOTE: using docker/rkt in our nodes we run out of space quick
+	// TODO: can this be moved to the runtime functions below?
+	cfg.Machines = make([]MachineConfiguration, cfg.Nodes)
+	for i := 0; i < cfg.Nodes; i++ {
+		cfg.Machines[i].Name = MachineName(i)
+		mountPath := path.Join(cfg.KubeSpawnDir, cfg.Name, cfg.Machines[i].Name, "mount")
+
+		var pm Pathmap
+		switch cfg.ContainerRuntime {
+		case RuntimeDocker:
+			pm = Pathmap{Dst: "/var/lib/docker", Src: mountPath}
+		case RuntimeRkt:
+			pm = Pathmap{Dst: "/var/lib/rktlet", Src: mountPath}
+		case RuntimeCrio:
+			pm = Pathmap{Dst: "/var/lib/containers", Src: mountPath}
+		}
+		cfg.Machines[i].Bindmount.ReadWrite = append(cfg.Machines[i].Bindmount.ReadWrite, pm)
+
+		// create dirs from above if they don't exist already
+		// TODO: should we create the dirs from here or move this to the check pkg
+		for _, pm := range cfg.Machines[i].Bindmount.ReadWrite {
+			if !fs.Exists(pm.Src) {
+				if err := fs.CreateDir(pm.Src); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func SetDefaults_DockerRuntime(cfg *ClusterConfiguration) error {
+	var err error
+	cfg.RuntimeConfiguration.Endpoint = DefaultDockerRuntimeEndpoint
+	// note: cgroup driver defaults to systemd on most systems, but there's
+	// an issue of runc <=1.0.0-rc2 that conflicts with --cgroup-driver=systemd,
+	// so we should use legacy driver "cgroupfs".
+	cfg.RuntimeConfiguration.UseLegacyCgroupDriver = true
+	return err
+}
+
+func SetDefaults_RktRuntime(cfg *ClusterConfiguration) error {
+	var err error
+	cfg.RuntimeConfiguration.Endpoint = DefaultRktRuntimeEndpoint
+
+	if cfg.RuntimeConfiguration.Rkt.RktBin == "" {
+		cfg.RuntimeConfiguration.Rkt.RktBin, err = exec.LookPath("rkt")
+		if err != nil {
+			return err
+		}
+	}
+	if cfg.RuntimeConfiguration.Rkt.RktletBin == "" {
+		cfg.RuntimeConfiguration.Rkt.RktletBin, err = exec.LookPath("rktlet")
+		if err != nil {
+			return err
+		}
+	}
+	if cfg.RuntimeConfiguration.Rkt.Stage1Image == "" {
+		cfg.RuntimeConfiguration.Rkt.Stage1Image = DefaultRktStage1ImagePath
+	}
+
+	cniPath := os.Getenv("CNI_PATH")
+	if cniPath == "" {
+		return errors.New("CNI_PATH was not set")
+	}
+	pms := []Pathmap{
+		{Dst: "/usr/bin/rkt", Src: cfg.RuntimeConfiguration.Rkt.RktBin},
+		{Dst: "/usr/bin/rktlet", Src: cfg.RuntimeConfiguration.Rkt.RktletBin},
+		{Dst: "/usr/bin/stage1-coreos.aci", Src: cfg.RuntimeConfiguration.Rkt.Stage1Image},
```

Comment:
can we change this so it works if I use a non-coreos stage1? (see https://github.com/kinvolk/kube-spawn/pull/162)

Reply:
forgot about this, thanks